operator/internal/controller/pulumi/session_test.go

```diff
@@ -357,17 +359,17 @@ func TestSetupWorkspace(t *testing.T) {
 						},
 					},
 				},
-				WorkspaceTemplate: &autov1alpha1.EmbeddedWorkspaceTemplateSpec{
-					Metadata: autov1alpha1.EmbeddedObjectMeta{
+				WorkspaceTemplate: &shared.WorkspaceApplyConfiguration{
+					ObjectMetaApplyConfiguration: &metav1apply.ObjectMetaApplyConfiguration{
 						Labels: map[string]string{
 							"custom": "label",
 						},
 					},
-					Spec: &autov1alpha1.WorkspaceSpec{
-						Image:              "custom-image",
-						ServiceAccountName: "custom-service-account",
-						PodTemplate: &autov1alpha1.EmbeddedPodTemplateSpec{
-							Metadata: autov1alpha1.EmbeddedObjectMeta{
+					Spec: &apply.WorkspaceSpecApplyConfiguration{
+						Image:              ptr.To("custom-image"),
+						ServiceAccountName: ptr.To("custom-service-account"),
+						PodTemplate: &apply.EmbeddedPodTemplateSpecApplyConfiguration{
+							Metadata: &apply.EmbeddedObjectMetaApplyConfiguration{
```

Comment:
If we're happy with the direction of this I can take it further and also use an apply config for our pod template -- it should be very minor.

Reply:
That would be pretty wonderful.

Reply:
I spoke too soon -- this is actually quite involved. I'll put up a WIP with what I have so far.